core-java-modules/core-java-io-5/src/test/java/com/baeldung/charstreams/CharStreamsUnitTest.java

```diff
@@ -0,0 +1,116 @@
+package com.baeldung.charstreams;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.*;
+import java.nio.file.*;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.jupiter.api.*;
+
+public class CharStreamsUnitTest {
+
+    private static final DailyTodo TODO = new DailyTodo("Code", LocalDateTime.now(), true);
+    private static final DailyTodo TODO_1 = new DailyTodo("Sleep", LocalDateTime.now(), false);
+    private static final DailyTodo TODO_2 = new DailyTodo("Read", LocalDateTime.now(), false);
+    public static String TODO_LIST_PW_FILE = "src/test/resources/pw_todos.txt";
+    public static String TODO_LIST_FW_FILE = "src/test/resources/fw_todos.txt";
+    
+
+    @BeforeEach
+    public void createFiles() throws IOException {
+        Files.createDirectories(Paths.get(TODO_LIST_PW_FILE).getParent());
+        Files.createDirectories(Paths.get(TODO_LIST_FW_FILE).getParent());
+    }
+
+    @Test
+    public void whenUsingTextData_thenPrintWriterWritesToFile() throws IOException {
+        try {
+            pwFile = CharStreams.writingStringToFilePrintWriter(TODO_LIST_PW_FILE , TODO.toString());
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO.toString());
+        }catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingTextData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingStringToFileFileWriter(TODO_LIST_FW_FILE, TODO.toString());
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+        
+    }
+
+    @Test
+    public void whenUsingFormattedData_thenPrintWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingNonStringToFilePrintWriter(TODO_LIST_PW_FILE, TODO_1);
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO_1.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingFormattedData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingNonStringToFileFileWriter(TODO_LIST_FW_FILE, TODO_1);
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO_1.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenWritingToExixtingFile_thenPrintWriterAppendsToFile() throws IOException {
+        try {
+            CharStreams.appendingToFilePrintWriter(TODO_LIST_PW_FILE, TODO_2);
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO_2.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }

```

Comment:
For these two append tests, to truly test append you need to first begin the test with an existing file that has existing content. Then, after the append, we should assert that both the previous content and the appended content is present.

NOTE: when I tested this with PrintWriter, it does not append in the way you assumed it worked. In the following code, the original file is overwritten:
```
        String existingText = "existing text";
        Files.write(Paths.get(TODO_LIST_PW_FILE), existingText.getBytes(StandardCharsets.UTF_8), StandardOpenOption.CREATE_NEW);

        try (PrintWriter printWriter = new PrintWriter(new File(TODO_LIST_PW_FILE), StandardCharsets.UTF_8)) {
            printWriter.append(TODO.toString());
        }

        assertThat(new File(TODO_LIST_PW_FILE)).hasContent(existingText + TODO.toString());
```

One correct way to append to an existing file using a PrintWriter is:
```
        String existingText = "existing text";
        Files.write(Paths.get(TODO_LIST_PW_FILE), existingText.getBytes(StandardCharsets.UTF_8), StandardOpenOption.CREATE_NEW);

        try (PrintWriter printWriter = new PrintWriter(new FileOutputStream(TODO_LIST_PW_FILE, true), false, StandardCharsets.UTF_8)) {
            printWriter.append(TODO.toString());
        }

        assertThat(new File(TODO_LIST_PW_FILE)).hasContent(existingText + TODO.toString());
```

Reply:
I added a new file creation with the first two lines in CharStreams.java: 
`public static void appendingToFilePrintWriter(String fileName, DailyTodo content) throws IOException {
        String existingText = "existing text";
        Files.write(Paths.get(fileName), existingText.getBytes(StandardCharsets.UTF_8), StandardOpenOption.CREATE_NEW);
        PrintWriter printWriter = new PrintWriter(new File(fileName), StandardCharsets.UTF_8);
        printWriter.append(content.toString());`

but it now fails.

I did something similar for the FileWriter append method.

Reply:
As mentioned above, the cause of the issue is constructing the writer not in append mode, see the second example above for how to specify append mode